runtime/runtime_storage.go

```diff
@@ -106,6 +121,23 @@ func (s *runtimeStorage) valueExists(
 	return exists
 }
 
+
+func (s *runtimeStorage) Read(_ *Interpreter, address common.Address, key string) OptionalValue {
+	// TODO (Ramtin) what to do with deferred
+	storable, ok := s.readValue(address, key, false)
```

Comment:
I was not sure about what to set for deferred here. I assume we need to support passing it.